### DISCUSSION TITLE
Should `refetchOnWindowFocus` be true globally?

### DISCUSSION THREAD

**Discussion by satejbidvai** [TanStack/query#7670] [Q&A] [answered]
In large codebases, I've seen `refetchOnWindowFocus` turned off by default.

When users return to a tab after a long time, seeing the latest data is still a good experience. 
For e.g., Let's say a page renders a list view or a list of notifications; when a user returns after a couple of hours, and the data has been updated, it's a good UX to re-fetch the list in the background.

I tried googling for any blogs/articles where I can find if it's good practice to keep the `refetchOnWindowFocus` on / off.

As the default decided by the React Query team is ON, there must be good reasons for that, and I wanted to understand them as I don't see them mentioned in the docs - just that one can disable it globally:

> [You can disable this globally or per-query using the refetchOnWindowFocus option ](https://tanstack.com/query/latest/docs/framework/react/guides/window-focus-refetching#disabling-per-query)

**Comment by 87xie** [comment]
Before TanStack Query v5, some unexpected refetches, detailed in PR #4805, likely led to the default setting of `refetchOnWindowFocus` being turned off to avoid unnecessary data fetching.

Providing a good user experience out of the box is important from a library perspective. Therefore, having `refetchOnWindowFocus` default to `true` ensures that users see the most up-to-date data when they return to a page.

That's my point, but I'm unsure about the actual considerations.

**Comment by TkDodo** [comment, marked as answer]
The idea is to have all the flags on per default and then have the user define a `staleTime` for their resources because all the refetches respect `staleTime`.
